### PR TITLE
Update tree-sitter to 0.22.6

### PR DESCRIPTION
**Pull request by eed3si9n**
Fixes https://github.com/tree-sitter/tree-sitter-scala/issues/384
Ref https://github.com/neovim/neovim/pull/28643

### CODE REVIEW

File: Cargo.toml
Comment (eed3si9n) [inline Cargo.toml:1]: Actual file I changed.

File: package.json [hunk 1]
```diff
@@ -3,23 +3,44 @@
...
+  },
+  "peerDependenciesMeta": {
+    "tree_sitter": {
+      "optional": true
+    }
   },
   "devDependencies": {
-    "tree-sitter-cli": "0.20.7",
-    "prettier": "3.0.0-alpha.6"
+    "tree-sitter-cli": "0.22.1",
```
Comment (eed3si9n) [inline package.json:34]: I think I changed this, and the rest of the change was made automatically.

File: bindings/rust/lib.rs [hunk 2]
```diff
@@ -6,7 +6,7 @@
 //! ```
 //! let code = "";
 //! let mut parser = tree_sitter::Parser::new();
-//! parser.set_language(tree_sitter_scala::language()).expect("Error loading scala grammar");
+//! parser.set_language(&tree_sitter_scala::language()).expect("Error loading scala grammar");
```
Comment (eed3si9n) [inline bindings/rust/lib.rs:9]: Minor Rust update.

**Comment by look** [comment]
Hello from the GitHub code search team. I would like to propose that this change be merged and a new release of the Rust crate be cut. Doing so will help us with an ongoing upgrade to our code navigation systems.

I see these changes are rolled into https://github.com/tree-sitter/tree-sitter-scala/pull/398, but the work there is ongoing. A smaller release that focuses simply on updating the tree-sitter dependency seems reasonable.

🙇

**Comment by eed3si9n** [comment]
> Hello from the GitHub code search team. I would like to propose that this change be merged and a new release of the Rust crate be cut. Doing so will help us with an ongoing upgrade to our code navigation systems.

👋🏼 . sounds good to me. I'll cut a release once I can land this PR.

**Review by ckipp01 — APPROVED** [review]

**Comment by eed3si9n** [comment]
@look https://crates.io/crates/tree-sitter-scala 0.22.0 is released

**Comment by look** [comment]
Thank you! Your work is appreciated 🙇

**Pull request merged**